apps/prairielearn/src/lib/question.js

```diff
@@ -1112,22 +1117,47 @@ function testQuestion(
           options,
           require_open,
           (err, ret_variant) => {
+            const generateEnd = Date.now();
+            generateDuration = generateEnd - generateStart;
             if (ERR(err, callback)) return;
             variant = ret_variant;
             debug('_testQuestion()', 'created variant_id: :', variant.id);
             callback(null);
           },
         );
       },
+      (callback) => {
+        const renderStart = Date.now();
+        getAndRenderVariant(
```

Comment:
Not really sure if this is doable or not, just raising for consideration: this calculation does not account for dynamic images that may take time to load (or other items loaded asynchronously). Should it? Again, this may not be easy and may not be the focus of the question.

Reply:
This is not straightforward to do and I'm entirely comfortable not trying to do that for now.